src/devices/m3ibike/m3ibike.cpp

```diff
@@ -589,12 +595,11 @@ bool m3ibike::isCorrectUnit(const QBluetoothDeviceInfo &device) {
         QSettings settings;
         keiser_m3i_out_t k3;
         int id = settings.value(QZSettings::m3i_bike_id, QZSettings::default_m3i_bike_id).toInt();
-        QHash<quint16, QByteArray> datas = device.manufacturerData();
-        QHashIterator<quint16, QByteArray> i(datas);
-        while (i.hasNext()) {
-            i.next();
-            if (parse_data(i.value(), &k3) && (!valid_id(id) || k3.system_id == id)) {
-                return true;
+        QMultiHash<quint16, QByteArray> datas = device.manufacturerData();
+        QHash<quint16, QByteArray> uniqueDatas;
+        for (auto it = datas.begin(); it != datas.end(); ++it) {
+            if (!uniqueDatas.contains(it.key())) {
+                uniqueDatas[it.key()] = it.value();
```

Comment:
After this change, the function never returns true for QT_VERSION >= 5.12.0, so the M3I bike is not detected.

Reply:
...thereby causing a test failure:
![image](https://github.com/user-attachments/assets/735d547e-171b-4c4e-a908-9064f36de358)

Reply:
While I don't have an M2I bike to test it on, assuming the scenario the test sets up is still valid, you would need to put back the check of the values:

```
for (auto it = datas.begin(); it != datas.end(); ++it) {
            if (!uniqueDatas.contains(it.key())) {
                uniqueDatas[it.key()] = it.value();
				
                if (parse_data(it.value(), &k3) && (!valid_id(id) || k3.system_id == id)) {
                    return true;
                }
            }
        }
```

But I'm not sure why you are trying to avoid duplicate keys? The keys are unique already, but I see that in Qt6 this data is now in a QMultiHash so there can be multiple values for each key. To access all the values for each key you need use datas.values(key).

Reply:
done thanks!